wagtail/tests/test_revision_model.py

```diff
@@ -237,3 +239,101 @@ def test_revision_cascade_on_object_delete(self):
                 self.assertEqual(Revision.objects.filter(**query).first(), revision)
                 instance.delete()
                 self.assertIs(Revision.objects.filter(**query).exists(), False)
+
+    def test_overwrite_revision(self):
+        self.instance.text = "Existing revision"
+        revision1 = self.instance.save_revision()
+        self.assertEqual(self.instance.revisions.count(), 1)
+        self.instance.text = "Updated revision"
+        revision2 = self.instance.save_revision(overwrite_revision=revision1)
+        self.assertEqual(self.instance.revisions.count(), 1)
+        self.assertEqual(revision1.id, revision2.id)
+        revision1.refresh_from_db()
+        self.assertEqual(revision1.content["text"], "Updated revision")
+
+    def test_cannot_overwrite_revision_that_is_not_latest(self):
+        self.instance.text = "Existing revision"
+        revision1 = self.instance.save_revision()
+        self.assertEqual(self.instance.revisions.count(), 1)
+        self.instance.text = "Second revision"
+        self.instance.save_revision()
+        self.assertEqual(self.instance.revisions.count(), 2)
+        self.instance.text = "Updated revision"
+        with self.assertRaises(PermissionDenied):
+            self.instance.save_revision(overwrite_revision=revision1)
+
+        self.assertEqual(self.instance.revisions.count(), 2)
+        latest_revision = self.instance.get_latest_revision()
+        self.assertEqual(latest_revision.content["text"], "Second revision")
+
+    def test_cannot_overwrite_revision_from_other_instance(self):
+        self.instance.text = "Existing revision"
+        self.instance.save_revision()
+
+        other_instance = RevisableModel.objects.create(text="other")
+        other_instance.text = "Existing other revision"
+        revision1 = other_instance.save_revision()
+
+        self.instance.text = "Updated revision"
+        with self.assertRaises(PermissionDenied):
+            self.instance.save_revision(overwrite_revision=revision1)
```

Comment:
I'm guessing this is just because the ID of `self.instance.latest_revision` isn't the same as `revision1`, not necessarily because of the mismatching models.

It's probably possible to do this if for some reason the instance's `latest_revision_id` on the database points to a `Revision` of a different model's instance... but if that happened, you'd have bigger problems to solve 😂